src/interchange/src/avro.rs

```diff
@@ -798,7 +799,33 @@ impl<'a> AvroDecode for AvroFlatDecoder<'a> {
         *self.buf = str_buf;
         Ok(())
     }
-    define_unexpected! {map}
+    #[inline]
+    fn map<A: AvroMapAccess>(self, a: &mut A) -> Result<Self::Out, AvroError> {
+        let mut map_buf = std::mem::take(self.buf);
```

Comment:
There's no reason to take this (and put it back on line 826), now that you're not passing it to the inner decoder.

Reply:
Good catch - thanks!